### PR TITLE
New commands, event listeners, refactoring

### PR DESCRIPTION
**Pull request by chives**
- Change password console command
- Activate/deactivate console command
- Saving last login for the user
- Separated templates for changing passowrd during resetting and activation

### CODE REVIEW

File: Command/ActivateUserCommand.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+EOT
+            );
+    }
+
+    /**
+     * @see Command
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $email      = $input->getArgument('email');
```
Comment (szymach) [inline Command/ActivateUserCommand.php]: excessive spaces

File: Command/ChangePasswordCommand.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            );
+    }
+
+    /**
+     * @see Command
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $email      = $input->getArgument('email');
+        $password   = $input->getArgument('password');
```
Comment (szymach) [inline Command/ChangePasswordCommand.php]: again, unnecessary spaces

File: Command/ChangePasswordCommand.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+        $userRepository = $this->getContainer()->get('admin_security.repository.user');
+        $user = $userRepository->findUserByEmail($email);
+        if (!($user instanceof UserPasswordChangeInterface)) {
```
Comment (szymach) [inline Command/ChangePasswordCommand.php]: I don't see the UserPasswordChangeInterface neither in this PR or the current master branch

Comment (szymach) [inline Command/ChangePasswordCommand.php]: Also, if this is an interface for the User, I'd name it something like UserWithChangeablePasswordInterface, current name suggests that it's for a class containing the password to be changed or a service doing the change. Perphaps even something like CanChangePasswordInterface would do.

File: Command/DeactivateUserCommand.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+EOT
+            );
+    }
+
+    /**
+     * @see Command
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $email      = $input->getArgument('email');
```
Comment (szymach) [inline Command/DeactivateUserCommand.php]: spaces

File: Command/ActivateUserCommand.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        $userRepository = $this->getContainer()->get('admin_security.repository.user');
+        $user = $userRepository->findUserByEmail($email);
+        if (!($user instanceof UserActivableInterface)) {
```
Comment (szymach) [inline Command/ActivateUserCommand.php]: I can't find this interface anywhere either

File: EventListener/LogoutUserListener.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace FSi\Bundle\AdminSecurityBundle\EventListener;
+
+use FSi\Bundle\AdminSecurityBundle\Event\AdminSecurityEvents;
+use FSi\Bundle\AdminSecurityBundle\Event\ChangePasswordEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class LogoutUserListener implements EventSubscriberInterface
```
Comment (szymach) [inline EventListener/LogoutUserListener.php]: maybe UserLogoutListener?

File: Resources/doc/events.md [hunk 7]
```diff
@@ -8,14 +8,14 @@ FSiAdminSecurityBundle provides some useful events which cause some default acti
 user instance. Its build-in subscribers:
```
Comment (szymach) [inline Resources/doc/events.md]: "*is fired when new password is set on user instance. Its build-in subscribers: " => "is fired when a new password is set on user's instance. It has following subscribers built-in :"

File: Resources/doc/events.md [hunk 8]
```diff
@@ -8,14 +8,14 @@ FSiAdminSecurityBundle provides some useful events which cause some default acti
...
 
 - [encode the new password](EventListener/UserEncodepasswordListener.php)
-- [ensure user is no longer enforced to change password if it was before](EventListener/UserPasswordChangedListener.php)
+- [ensure user is no longer enforced to change password if it was before](EventListener/ClearChangePasswordEnforcementListener.php)
 
 ## Password reset request event
 
 ``FSi\Bundle\AdminSecurityBundle\Event\AdminSecurityEvents::RESET_PASSWORD_REQUEST`` is fired when user requested to
 reset his/her password. Its built-in subscribers:
```
Comment (szymach) [inline Resources/doc/events.md]: "is fired when user requested to reset his/her password. Its built-in subscribers" => "is fired when user requests a password reset. It has following built-in subscribers"

File: Resources/doc/events.md [hunk 9]
```diff
@@ -23,7 +23,7 @@ reset his/her password. Its built-in subscribers:
 from the console command. Its build-in subscribers:
```
Comment (szymach) [inline Resources/doc/events.md]: "is fired when new user" => "is fired when a new user"

File: Resources/doc/events.md [hunk 10]
```diff
@@ -35,4 +35,4 @@ Its build-in subscribers:
 ## Other listeners 
 
 There is also one subscriber which subscribes all of the above events and is responsible for saving changes to the
-corresponding ORM/ODM - [DoctrineUserListener](EventListener/DoctrineUserListener.php) by the ODM/ORM.
+corresponding ORM/ODM - [PersistDoctrineUserListener](EventListener/PersistDoctrineUserListener.php) by the ODM/ORM.
```
Comment (szymach) [inline Resources/doc/events.md]: "There is also one subscriber which subscribes all of the above events and is" => "There is also a subscriber for all of the above events and it is"

File: EventListener/LogoutUserListener.php (hunk 6 above)
Comment (chives) [inline EventListener/LogoutUserListener.php]: All listeners have their names indicating what they do. Their names are not indicating which events they subscribe.

**Comment by rn0** [comment]
:+1:

**Comment by JarekW** [comment]
:+1:

**Comment by szymach** [comment]
:+1:

**Pull request merged**